infra/graph_generators.py

```diff
@@ -0,0 +1,182 @@
+#!/usr/bin/env python3
+"""
+This Module provides functions for generating simple networks
+"""
+import networkx as nx
+
+
+def bidirectional(graph):
+    """
+    Returns a bidirectional copy of the input graph.
+
+    For every edge in the input graph, the bidirectional graph
+    has a corresponding reversed edge.
+
+    Does not mutate the input graph.
+    """
+    output = nx.MultiDiGraph()
+    for e in graph.edges:
+        output.add_edge(e[0], e[1])
+        output.add_edge(e[1], e[0])
+    return output
+
+
+def reverse(graph):
+    """
+    Returns a reversed direction copy of the input graph.
+
+    For every edge in the input graph, there will be a reversed edge in the
+    resultant graph.
+
+    Does not mutate the input graph.
+    """
+    output = nx.MultiDiGraph()
+    for e in graph.edges:
+        output.add_edge(e[1], e[0])
+    return output
+
+
+def set_types(graph, node_types="vanilla", edge_types="vanilla"):
```

Comment:
I don't like this API (or the way that types are handled in this PR generally). When node_types is provided as a dict, there's no guarantee that every node is present in the dict, so I expect that often times nodes will go without any type at all. So there's no guarantee that types are consistently present. Also, AFAIK we don't have any code yet that actually consumes node types.

Can we strip out the node and edge types concept from this PR and merge without them? Then, once we actually need node types, we can put them back in, but we will know a little more at that time based on what the actual usage is.